sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/binaryfile/BinaryFileFormat.scala

```diff
@@ -101,43 +98,27 @@ class BinaryFileFormat extends FileFormat with DataSourceRegister {
     val filterFuncs = filters.map(filter => createFilterFunction(filter))
 
     file: PartitionedFile => {
-      val path = file.filePath
-      val fsPath = new Path(path)
-
+      val path = new Path(file.filePath)
       // TODO: Improve performance here: each file will recompile the glob pattern here.
-      if (pathGlobPattern.forall(new GlobFilter(_).accept(fsPath))) {
-        val fs = fsPath.getFileSystem(broadcastedHadoopConf.value.value)
-        val fileStatus = fs.getFileStatus(fsPath)
-        val length = fileStatus.getLen
-        val modificationTime = fileStatus.getModificationTime
-
-        if (filterFuncs.forall(_.apply(fileStatus))) {
-          val stream = fs.open(fsPath)
-          val content = try {
-            ByteStreams.toByteArray(stream)
-          } finally {
-            Closeables.close(stream, true)
-          }
-
-          val fullOutput = dataSchema.map { f =>
-            AttributeReference(f.name, f.dataType, f.nullable, f.metadata)()
-          }
-          val requiredOutput = fullOutput.filter { a =>
-            requiredSchema.fieldNames.contains(a.name)
+      if (pathGlobPattern.forall(new GlobFilter(_).accept(path))) {
```

Comment:
Can we make it one if-else while we're here? For instance,

```scala
val isPatternMatched = pathGlobPattern.forall(new GlobFilter(_).accept(fsPath))

// These vals are intentionally lazy to avoid unnecessary file access via short-circuiting. 
lazy val fs = fsPath.getFileSystem(broadcastedHadoopConf.value.value)
lazy val fileStatus = fs.getFileStatus(fsPath)
lazy val shouldNotFilterOut = filterFuncs.forall(_.apply(fileStatus))
      
if (isPatternMatched && shouldNotFilterOut) {
  ...
  Iterator(requiredColumns(internalRow))
} else {
  Iterator.empty[InternalRow]
}
```

Reply:
In the current impl, `filterFuncs` are not touched if path doesn't match.

Reply:
Yes, this suggestion does not touch both because they are lazy